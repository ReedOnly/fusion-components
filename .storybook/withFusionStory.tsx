import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
