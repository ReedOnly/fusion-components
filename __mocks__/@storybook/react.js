import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
