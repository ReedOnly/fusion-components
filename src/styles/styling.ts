export default {
    grid: (value: number) => `calc(var(--grid-unit) * ${value}px)`,

    // Colors
    cssColors: {
        primary: 'var(--color-primary)',
        primaryAlt1: 'var(--color-primary-alt1)',
        primaryAlt2: 'var(--color-primary-alt2)',
        primaryAlt3: 'var(--color-primary-alt3)',
        primaryAlt4: 'var(--color-primary-alt4)',
        primaryAccent: 'var(--color-primary-accent)',
        primaryHover: 'var(--color-primary-hover)',
        primaryHoverAlt1: 'var(--color-primary-hover-alt1)',
        secondary: 'var(--color-secondary)',
        secondaryAlt1: 'var(--color-secondary-alt1)',
        secondaryAlt2: 'var(--color-secondary-alt2)',
        secondaryAlt3: 'var(--color-secondary-alt3)',
        secondaryAlt4: 'var(--color-secondary-alt4)',
        secondaryAccent: 'var(--color-secondary-accent)',
        black: 'var(--color-black)',
        blackAlt1: 'var(--color-black-alt1)',
        blackAlt2: 'var(--color-black-alt2)',
        blackAlt3: 'var(--color-black-alt3)',
        blackAlt4: 'var(--color-black-alt4)',
        blackAlt5: 'var(--color-black-alt5)',
        white: 'var(--color-white)',
        supplementary: 'var(--color-supplementary)',
        supplementaryAlt1: 'var(--color-supplementary-alt1)',
        supplementaryAlt2: 'var(--color-supplementary-alt2)',
        yellow: 'var(--color-yellow)',
        orange: 'var(--color-orange)',
        red: 'var(--color-red)',
        green: 'var(--color-green)',
        purple: 'var(--color-purple)',
        blue: 'var(--color-blue)',
        turquoise: 'var(--color-turquoise)',
    }
};
