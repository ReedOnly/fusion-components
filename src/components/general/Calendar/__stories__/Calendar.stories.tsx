import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../../.storybook/withFusionStory';

import Calendar from '../index';
import { Month } from '@equinor/fusion';
import styling from 'styles/styling';

const stories = storiesOf('General|Calendar', module);
stories.addDecorator(withFusionStory('Calendar'));

const initialYear = new Date().getFullYear();
const initialMonth = new Date().getMonth() as Month;

stories.add('Default', () => (
    <div style={{ width: styling.grid(7 * 6), margin: '0 auto' }}>
        <Calendar initialYear={initialYear} initialMonth={initialMonth} />
    </div>
));

const InteractiveCalendar = () => {
    const [selectedDate, setSelecteDate] = useState<Date>();
    return (
        <div style={{ width: styling.grid(7 * 6), margin: '0 auto' }}>
            <Calendar
                initialYear={initialYear}
                initialMonth={initialMonth}
                interactive
                onChange={setSelecteDate}
                selectedDate={selectedDate}
            />
        </div>
    );
};

stories.add('Interactive', () => <InteractiveCalendar />);
