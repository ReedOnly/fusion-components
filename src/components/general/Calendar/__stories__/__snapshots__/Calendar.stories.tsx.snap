// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|Calendar Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "width": "calc(var(--grid-unit) * 42px)",
        }
      }
    >
      <div
        className="container comfortable"
      >
        <header
          className="header"
        >
          <button
            className="container comfortable"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="iconContainer"
            >
              <svg
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M15 6.70508C14.8132 6.51783 14.5595 6.4126 14.295 6.4126C14.0305 6.4126 13.7769 6.51783 13.59 6.70508L9.00002 11.2951C8.61002 11.6851 8.61002 12.3151 9.00002 12.7051L13.59 17.2951C13.98 17.6851 14.61 17.6851 15 17.2951C15.39 16.9051 15.39 16.2751 15 15.8851L11.12 11.9951L15 8.11508C15.39 7.72508 15.38 7.08508 15 6.70508Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <h4>
            September 2020
          </h4>
          <button
            className="container comfortable"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="iconContainer"
            >
              <svg
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M9.00002 6.71051C8.61002 7.10051 8.61002 7.73051 9.00002 8.12051L12.88 12.0005L9.00002 15.8805C8.61002 16.2705 8.61002 16.9005 9.00002 17.2905C9.39002 17.6805 10.02 17.6805 10.41 17.2905L15 12.7005C15.39 12.3105 15.39 11.6805 15 11.2905L10.41 6.70051C10.03 6.32051 9.39002 6.32051 9.00002 6.71051Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </header>
        <div
          className="grid"
        >
          <header>
            Mon
          </header>
          <header>
            Tue
          </header>
          <header>
            Wed
          </header>
          <header>
            Thu
          </header>
          <header>
            Fri
          </header>
          <header>
            Sat
          </header>
          <header>
            Sun
          </header>
          <span
            className="day notSelectedMonth"
          >
            31
          </span>
          <span
            className="day"
          >
            01
          </span>
          <span
            className="day"
          >
            02
          </span>
          <span
            className="day"
          >
            03
          </span>
          <span
            className="day"
          >
            04
          </span>
          <span
            className="day"
          >
            05
          </span>
          <span
            className="day"
          >
            06
          </span>
          <span
            className="day"
          >
            07
          </span>
          <span
            className="day"
          >
            08
          </span>
          <span
            className="day isToday"
          >
            09
          </span>
          <span
            className="day"
          >
            10
          </span>
          <span
            className="day"
          >
            11
          </span>
          <span
            className="day"
          >
            12
          </span>
          <span
            className="day"
          >
            13
          </span>
          <span
            className="day"
          >
            14
          </span>
          <span
            className="day"
          >
            15
          </span>
          <span
            className="day"
          >
            16
          </span>
          <span
            className="day"
          >
            17
          </span>
          <span
            className="day"
          >
            18
          </span>
          <span
            className="day"
          >
            19
          </span>
          <span
            className="day"
          >
            20
          </span>
          <span
            className="day"
          >
            21
          </span>
          <span
            className="day"
          >
            22
          </span>
          <span
            className="day"
          >
            23
          </span>
          <span
            className="day"
          >
            24
          </span>
          <span
            className="day"
          >
            25
          </span>
          <span
            className="day"
          >
            26
          </span>
          <span
            className="day"
          >
            27
          </span>
          <span
            className="day"
          >
            28
          </span>
          <span
            className="day"
          >
            29
          </span>
          <span
            className="day"
          >
            30
          </span>
          <span
            className="day notSelectedMonth"
          >
            01
          </span>
          <span
            className="day notSelectedMonth"
          >
            02
          </span>
          <span
            className="day notSelectedMonth"
          >
            03
          </span>
          <span
            className="day notSelectedMonth"
          >
            04
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|Calendar Interactive 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "width": "calc(var(--grid-unit) * 42px)",
        }
      }
    >
      <div
        className="container comfortable"
      >
        <header
          className="header"
        >
          <button
            className="container comfortable"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="iconContainer"
            >
              <svg
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M15 6.70508C14.8132 6.51783 14.5595 6.4126 14.295 6.4126C14.0305 6.4126 13.7769 6.51783 13.59 6.70508L9.00002 11.2951C8.61002 11.6851 8.61002 12.3151 9.00002 12.7051L13.59 17.2951C13.98 17.6851 14.61 17.6851 15 17.2951C15.39 16.9051 15.39 16.2751 15 15.8851L11.12 11.9951L15 8.11508C15.39 7.72508 15.38 7.08508 15 6.70508Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <h4>
            September 2020
          </h4>
          <button
            className="container comfortable"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="iconContainer"
            >
              <svg
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M9.00002 6.71051C8.61002 7.10051 8.61002 7.73051 9.00002 8.12051L12.88 12.0005L9.00002 15.8805C8.61002 16.2705 8.61002 16.9005 9.00002 17.2905C9.39002 17.6805 10.02 17.6805 10.41 17.2905L15 12.7005C15.39 12.3105 15.39 11.6805 15 11.2905L10.41 6.70051C10.03 6.32051 9.39002 6.32051 9.00002 6.71051Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </header>
        <div
          className="grid"
        >
          <header>
            Mon
          </header>
          <header>
            Tue
          </header>
          <header>
            Wed
          </header>
          <header>
            Thu
          </header>
          <header>
            Fri
          </header>
          <header>
            Sat
          </header>
          <header>
            Sun
          </header>
          <span
            className="day notSelectedMonth"
          >
            <button
              className="container comfortable isDisabled"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                31
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                01
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                02
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                03
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                04
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                05
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                06
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                07
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                08
              </span>
            </button>
          </span>
          <span
            className="day isToday"
          >
            <button
              className="container comfortable isToggler"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                09
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                10
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                11
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                12
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                13
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                14
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                15
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                16
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                17
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                18
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                19
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                20
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                21
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                22
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                23
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                24
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                25
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                26
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                27
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                28
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                29
              </span>
            </button>
          </span>
          <span
            className="day"
          >
            <button
              className="container comfortable"
              disabled={false}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                30
              </span>
            </button>
          </span>
          <span
            className="day notSelectedMonth"
          >
            <button
              className="container comfortable isDisabled"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                01
              </span>
            </button>
          </span>
          <span
            className="day notSelectedMonth"
          >
            <button
              className="container comfortable isDisabled"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                02
              </span>
            </button>
          </span>
          <span
            className="day notSelectedMonth"
          >
            <button
              className="container comfortable isDisabled"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                03
              </span>
            </button>
          </span>
          <span
            className="day notSelectedMonth"
          >
            <button
              className="container comfortable isDisabled"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                04
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
